wiring/src/spark_wiring_ble.cpp

```diff
@@ -662,6 +662,10 @@ size_t BleAdvertisingData::appendCustomData(const uint8_t* buf, size_t len, bool
     return append(BleAdvertisingDataType::MANUFACTURER_SPECIFIC_DATA, buf, len, force);
 }
 
+size_t BleAdvertisingData::appendAppearance(ble_sig_appearance_t appearance) {
+    return append(BleAdvertisingDataType::APPEARANCE, (const uint8_t*)&appearance, 2, false);
```

Comment:
The last argument `false` specify that it will overwrite the existing appearance if exists, otherwise, append the appearance. So there will always be one appearance in either advertising data or scan response data. According to the BLE CSS, appearance shall not appear more than once in a block and shall not appear in both the AD and SRD of the same extended advertising interval, but this should be guarded by user application.

Reply:
So, the last argument should be `true`? While we can say that the user application should guard against this, I'd say if we can make sure that we are not going off spec, we should do that.

Reply:
`force = true` means that we will append the appearance anyway, regardless of the existence in current advertising data. Like what we've done for `appendLocalName()`.  The advertising data and scan response data can be set separately, the wiring layer cannot guard the conflict, as it doesn't cache the data. The only way is checking the data validity in BLE HAL and return error code. Do you think we should do that now or later?

Reply:
I'd say let's queue a story for later. This can be just setting a flag that a particular important piece of data (which should be singular) is already present in the advertisement packet and either overwrite it or return an error.